lib/Parser/ptree.h

```diff
@@ -494,6 +494,7 @@ class ParseNodeFnc : public ParseNode
     bool canBeDeferred;
     bool isBodyAndParamScopeMerged; // Indicates whether the param scope and the body scope of the function can be merged together or not.
                                     // We cannot merge both scopes together if there is any closure capture or eval is present in the param scope.
+    bool needsScopeObject;
```

Comment:
nit: should we move the all bools together (from line 464) for better packing?

Reply:
Done